### DISCUSSION TITLE
Introducing the PyRI Open Source Teach Pendant

### DISCUSSION THREAD

**Discussion by johnwason** [tesseract-robotics/tesseract#629] [General]
https://github.com/pyri-project/pyri-core/blob/master/README.md
https://github.com/pyri-project/pyri-tesseract-planning

The Python Restricted Industrial (PyRI) Open Source Teach Pendant is a high-level programming environment for Robot Raconteur and ROS ecosystems. It is programmed using Blockly, a restricted Python dialect, and a top level state machine. It has an extension available for Tesseract, for collision free planning. We just finished a funded project to do the initial development, but it is still at a prototype level and has a lot more work to be done before it is ready for general use. It is Apache 2.0 licensed.

https://www.youtube.com/watch?v=9KSYgGpG8mk

**Comment by gavanderhoorn** [comment]
Congratulations on completing the funded part :+1: 

Are there plans to develop the OSS TP further?

**Reply by johnwason** [reply]
Yeah, I am going to continue developing it as part of my company's suite. Contributions are of course welcome!

**Comment by johnwason** [comment]
Demo video: https://www.youtube.com/watch?v=9KSYgGpG8mk
